ocaml/xapi/xapi_ha_vm_failover.ml

```diff
@@ -318,7 +318,12 @@ let compute_max_host_failures_to_tolerate ~__context ?live_set ?protected_vms ()
   let protected_vms = match protected_vms with
     | None -> all_protected_vms ~__context
     | Some vms -> vms in
-  let nhosts = List.length (Db.Host.get_all ~__context) in
+  let total_hosts = List.length (Db.Host.get_all ~__context) in
+  (* For corosync HA less than half of the pool can fail whilst maintaining quorum *)
+  let corosync_ha_max_hosts = ((total_hosts - 1) / 2) in
+  let nhosts = match Db.Cluster.get_all ~__context with
+    | [] -> total_hosts
+    | _ -> corosync_ha_max_hosts in
```

Comment:
Could you make this call a function in Xapi_clustering.ml to calculate the failures to tolerate? As mentioned above it needs to know about enabled/disabled hosts.